### PR TITLE
Set up CI with Azure Pipelines

### PR DESCRIPTION
**Pull request by kkjasoncheung**
- Add CI for unit and GraphQL testing with Azure Pipelines
- Containerize flask app, psql database and newman

### CODE REVIEW

File: azure-pipelines.yml [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+
+pool:
+  vmImage: "ubuntu-latest"
+
+steps:
+  - task: UsePythonVersion@0
+    inputs:
+      versionSpec: "3.6"
+  - script: pip install -r requirements.txt
+    displayName: "Intall packages"
```
Comment (boiledbuns) [inline azure-pipelines.yml]: nit: typo

File: Dockerfile [hunk 2]
```diff
@@ -0,0 +1,16 @@
+# pull official base image
+FROM python:3.7
+
+RUN apt-get upgrade
+# set work directory
+WORKDIR /app
+# copy project
+COPY . .
+
+ENV FLASK_APP "stronk"
```
Comment (boiledbuns) [inline Dockerfile]: Is there a difference between setting up the env manually and running the `setup` script?

File: azure-pipelines.yml [hunk 3]
```diff
@@ -0,0 +1,43 @@
+# Starter pipeline
+# Start with a minimal pipeline that you can customize to build and deploy your code.
+# Add steps that build, run tests, deploy, and more:
+# https://aka.ms/yaml
+
+trigger:
+  - master
```
Comment (boiledbuns) [inline azure-pipelines.yml]: Should we add a PR trigger https://docs.microsoft.com/en-us/azure/devops/pipelines/customize-pipeline?view=azure-devops? It looks like the CI is being run right now, but I'm not sure how bc this isn't on master

**Review by boiledbuns — COMMENTED** [review]
Looks good, but I'm not super familiar with ci/cd. Should we have a separate docker compose file for each bc it looks like the current one does both which means the app is deployed w dev dep?

Also, I'm seeing the ci working, but the `DATABASE_URL` env var is not being set
https://dev.azure.com/kkjasoncheung/stronk/_build/results?buildId=84&view=results

File: Dockerfile (hunk 2 above)
Comment (kkjasoncheung) [reply Dockerfile]: Hmm true we can just use the `script.sh`

File: azure-pipelines.yml (hunk 3 above)
Comment (kkjasoncheung) [reply azure-pipelines.yml]: I think it's because I configured the CI to trigger on PR validation on Azure Pipelines. I can move that config to this file

**Comment by kkjasoncheung** [comment]
> Looks good, but I'm not super familiar with ci/cd. Should we have a separate docker compose file for each bc it looks like the current one does both which means the app is deployed w dev dep?
> 
> Also, I'm seeing the ci working, but the `DATABASE_URL` env var is not being set
> https://dev.azure.com/kkjasoncheung/stronk/_build/results?buildId=84&view=results

By "each" do you mean separate out the web, db and postman service into three docker compose files? If that's the case I don't think that'll benefit us all that much because we need to run these three services together for the test. If anything I can separate the web and db services into its one docker compose file and postman service into its own for testing/dev purposes and extend when needed.

So `docker-compose.yml` will contain web and db and `docker-compose.test.yml` will contain postman.

**Pull request merged**